things/api/things/http/endpoint.go

```diff
@@ -248,7 +250,7 @@ func createChannelEndpoint(svc things.Service) endpoint.Endpoint {
 			return nil, err
 		}
 
-		ch := things.Channel{Name: req.Name, Metadata: req.Metadata}
+		ch := things.Channel{Name: req.Name, ID: req.ID, Metadata: req.Metadata}
```

Comment:
Create a new test case instead of mofying previous ones. ID shouldn't be mandatory